internal/provider/data_source_gitlab_group_variables.go

```diff
@@ -0,0 +1,77 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_group_variables", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_group_variables`" + ` data source allows to retrieve all CI/CD variables in a group.
```

Comment:
I'd suggest using GitLab's term
```suggestion
		Description: `The ` + "`gitlab_group_variables`" + ` data source allows to retrieve all group-level CI/CD variables.
```